frontend/components/Company.vue

```diff
@@ -0,0 +1,205 @@
+                      >
+                        {{ company.salary_id }}
+                      </div>
+                    </div>
+                  </div>
+                </td>
+                <td class="px-6 py-4 whitespace-nowrap">
+                  <div
+                    class="text-sm text-gray-900"
+                    style="color: #000000; font-family: 'Inter', sans-serif"
+                  >
+                    {{ company.company_name }}
+                  </div>
+                </td>
+                <td class="px-6 py-4 whitespace-nowrap">
+                  <div
+                    class="text-sm text-gray-900"
+                    style="color: #000000; font-family: 'Inter', sans-serif"
+                  >
+                    {{ company.job_title }}
+                  </div>
+                </td>
+                <td class="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
+                  <div
+                    class="text-sm text-gray-900"
+                    style="color: #000000; font-family: 'Inter', sans-serif"
+                  >
+                    {{ company.salary }} FCFA
+                  </div>
+                </td>
+                <td
+                  class="px-6 py-4 whitespace-nowrap text-sm font-medium flex"
+                >
+                  <div class="flex" v-for="item in company.rating" :key="item">
+                    <img class="w-4 h-4 mr-1" :src="startPicture" />
+                  </div>
+                </td>
+                <td
+                  class="
+                    px-6
+                    py-4
+                    whitespace-nowrap
+                    text-right text-sm
+                    font-medium
+                  "
+                >
+                  <a
+                    @click="toggleAccordion()"
+                    class="flex items-center space-x-3 cursor-pointer"
+                    style="color: #000000; font-family: 'Inter', sans-serif"
+                    :aria-expanded="isOpen"
+                    :aria-controls="`collapse${_uid}`"
+                    >Details</a
+                  >
+                </td>
+              </tr>
+              <div
+                v-show="isOpen"
+                class="w-10/12 bg-red-500"
+                :id="`collapse${_uid}`"
+              >
+                <!-- <slot name="{{company.comment}}" /> -->
+                <p class="w-full bg-yellow-600">{{ company.comment }}</p>
+              </div>
+            </tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+import axios from 'axios'
+import Vue from 'vue'
+import { BASE_URL } from '../constants/api'
+export default Vue.extend({
+  name: 'CompanyComponent',
+  data() {
+    return {
+      companies: [],
+      isOpen: false,
+      startPicture: require('../assets/star.png'),
+    }
+  },
+  mounted() {
+    axios.get(BASE_URL + '/ratings').then((response) => {
```

Comment:
why are you fetching the data in the mounted() method??

in the vue Lifecycle mounted will be call after the DOM was already show to the user. it is recommended to fetch the data in the created methods. so that the data will be available when building the DOM. so i thin k it is better to fetch the data in the created method.
can you please change it??

Reply:
i think it will also be a good idea to catch the error that can occur when fetching the data from the backend.
so i recommend to also add a .catch((error) =>{...}) after the .then().

Reply:
I'm based on the official VueJS documentation, they used the "mount" lifecycle (https://fr.vuejs.org/v2/cookbook/using-axios-to-consume-apis.html), I believe that the ideal would be to do as you described, so that the data is available during the construction of the DOM. Thank you